crates/compiler/builtins/bitcode/src/crypto.zig

```diff
@@ -0,0 +1,91 @@
+const std = @import("std");
+const builtin = @import("builtin");
+const crypto = std.crypto;
+const sha2 = crypto.hash.sha2;
+const list = @import("list.zig");
+const utils = @import("utils.zig");
+const testing = std.testing;
+
+const Sha256 = extern struct {
+    location: *sha2.Sha256,
```

Comment:
A pointer because:

-  The native Zig sha256 object doesn't have a defined layout and so Zig makes it (at least) difficult to pass the actual bytes in and out through FFI.
-  Given the possible  use case of adding streaming data a few bytes at a time, it seems like a good candidate for opportunistic mutation